### PR TITLE
kernel: bump 6.6 to 6.6.57

### PR DESCRIPTION
**Pull request by graysky2**
1. Update target/linux/generic/config-6.6 for new ksym
2. Refresh patches

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.57

Added:
	generic/backport-6.6/777-netfilter-xtables-fix-typo-causing-some-targets-to-not-load-on-IPv6.patch[1]

Manually rebased:
	generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch

Removed upstreamed:
	gemini/patches-6.6/0001-net-ethernet-cortina-Drop-TSO-support.patch[2]
	gemini/patches-6.6/0004-net-ethernet-cortina-Restore-TSO-support.patch[3]

All other patches automatically rebased.

1. https://lore.kernel.org/all/20241019-xtables-typos-v2-1-6b8b1735dc8e@0upti.me/
2 https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.57&id=452c0740d72c6a77a41f6ddc318a48f18c3d2346
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.57&id=611f74b0e7fb93ee2366d9d7edca546806b220e9

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

### CODE REVIEW

File: target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch [hunk 1]
```diff
@@ -209,6 +208,7 @@ Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
 +		.targetsize     = sizeof(struct xt_connmark_tginfo3),
```
Comment (DragonBluep) [inline target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:208]: Please split this struct into two parts to follow the upstream changes:
```
+	{
+		.name           = "CONNMARK",
+		.revision       = 3,
+		.family         = NFPROTO_IPV4,
+		.checkentry     = connmark_tg_check,
+		.target         = connmark_tg_v3,
+		.targetsize     = sizeof(struct xt_connmark_tginfo3),
+		.destroy        = connmark_tg_destroy,
+		.me             = THIS_MODULE,
+ 	},
+#if IS_ENABLED(CONFIG_IP6_NF_IPTABLES)
+	{
+		.name           = "CONNMARK",
+		.revision       = 3,
+		.family         = NFPROTO_IPV6,
+		.checkentry     = connmark_tg_check,
+		.target         = connmark_tg_v3,
+		.targetsize     = sizeof(struct xt_connmark_tginfo3),
+		.destroy        = connmark_tg_destroy,
+		.me             = THIS_MODULE,
+ 	},
+#endif
```
ref: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.57&id=997f67d813ce0cf5eb3cdb8f124da68141e91b6c

Comment (graysky2) [reply target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:208]: Sorry, I do not understand what you're asking.

Comment (graysky2) [reply target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:208]: @DragonBluep - Please see my latest commit and let me know if it is correctly split.

Comment (graysky2) [reply target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:208]: yeah, I blew it.  Can you please provide me with a diff I can patch against my latest commit to fix it?

File: target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch [hunk 2]
```diff
@@ -209,6 +208,24 @@ Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
...
 +		.destroy        = connmark_tg_destroy,
 +		.me             = THIS_MODULE,
- 	}
++	};
```
Comment (namiltd) [inline target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:211]: change to:
```
+	},
```

Comment (graysky2) [reply target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch]: Thanks

**Comment by namiltd** [comment]
Tested on ramips/mt7620 TP-Link Archer C5 v4

**Review by namiltd — APPROVED** [review]

File: target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch [hunk 3]
```diff
@@ -209,6 +208,24 @@ Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
 +		.targetsize     = sizeof(struct xt_connmark_tginfo3),
```
Comment (DragonBluep) [inline target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:208]: change NFPROTO_UNSPEC to NFPROTO_IPV4

**Comment by DragonBluep** [comment]
And this patch need to be backported. https://lore.kernel.org/all/20241018-xtables-typos-v1-1-02a51789c0ec@0upti.me/

**Comment by DragonBluep** [comment]
And this patch need to be backported. https://lore.kernel.org/all/20241018-xtables-typos-v1-1-02a51789c0ec@0upti.me/

**Comment by namiltd** [comment]
Updated patch to https://lore.kernel.org/all/20241019-xtables-typos-v2-1-6b8b1735dc8e@0upti.me/
but it hasn't been approved anywhere yet.

File: target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch (hunk 3 above)
Comment (graysky2) [reply target/linux/generic/hack-6.6/645-netfilter-connmark-introduce-set-dscpmark.patch:208]: Done

**Comment by graysky2** [comment]
I added the v2 patch.  Will push update shortly.

**Comment by systemcrash** [comment]
@linusw those cortina patches were your fixes, ya? In case you're wondering why iperf3 always flatlines at 1MB/sec, it's stateless/connectionless (so no ACK), you need to manually assign a desired bandwidth, e.g. `-b 1000M`.

**Pull request merged**